5285 DAPI DTRYGDEAVTALE TOGGLE Endre sakstype og sakstema

janitamarielle opened the pull request:
- Skilte endreBehandlingModal i to: en der behandle_alle_saker toggle er på, og en der den er av (gammel).
- Den gamle oppdaterer via behandling/endre, mens den nye oppdaterer via fagsak/endre. 
- I begge trakk jeg inn nullstillFlyt siden vi trenger sjekk på når vi resetter trygdeavtale-flyt. 
- Vi trengte heller ikke hente behandling ++ på nytt siden dette skjer ved reload av siden eller tilAnnenSide. 
- I den nye versjonen fjernet jeg oppfriskSaksopplysninger siden dette skjer i backend.

Melosys-api: https://github.com/navikt/melosys-api/pull/1519
https://jira.adeo.no/browse/MELOSYS-5285

janitamarielle committed:
Skilte endreBehandlingModal i to: en der behandle_alle_saker toggle er på, og en der den er av (gammel).
Den nye oppdaterer via behandling/endre, mens den nye oppdaterer via fagsak/endre.
I begge trakk jeg inn nullstillFlyt siden vi trenger sjekk på når vi resetter trygdeavtale-flyt. Vi trengte heller ikke hente behandling ++ på nytt siden dette skjer ved reload av siden eller tilAnnenSide.
I den nye versjonen fjernet jeg oppfriskSaksopplysninger siden dette skjer i backend.

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-5285_endre-sakstype-sakstema

janitamarielle committed:
Småtteri

frans-a reviewed (commented):
Jeg åpnet en ny vurdering lokalt og jeg får lov å endre sakstype og sakstema

Stemmer ikke med:
>1: Gitt at jeg er i en pågående behandling
og behandlingen ikke er den første behandlingen i saken
skal feltene for sakstype og sakstema være disablet
og det skal stå en informasjonsmelding som sier "Du kan bare endre sakstype og -tema i den første behandlingen i saken"

janitamarielle committed:
Legger til saksnummer i henting av sakstyper og sakstemaer. Legger til feilmelding og disabler når sakstype og sakstema ikke kan endres - aka når behandling ikke er den eneste i fagsaken.

janitamarielle commented:
>Jeg åpnet en ny vurdering lokalt og jeg får lov å endre sakstype og sakstema

Skal ha blitt løst med [1379e53](https://github.com/navikt/melosys-web/pull/1703/commits/1379e533e2ad41dca862a904e9b2b98f42d83ad0)

janitamarielle committed:
Ikke reset trygdeavtale når det er tom flyt.

janitamarielle committed:
Bruker skalViseTomFlyt siden Trygdeavtale ikke kan ha SED

janitamarielle committed:
Bruker skalViseTomFlyt siden Trygdeavtale ikke kan ha SED

janitamarielle committed:
Vi kan ikke bruke muligeVerdierPlussValgt siden eksisterende verdi ikke alltid er lovlig verdi. Fjerner derfor bruken av denne.
Det medfører at saker med Endret Periode viser ENDRET_PERIODE istedenfor termen, men avklart med fag at dette er ok. Endret Periode er uansett ikke lengre støttet ved opprettelse.

janitamarielle committed:
Støtter at sakstype kan endres, men fortsatt være i tom flyt. Gjør denne logikken i behandling siden det gir mening å laste inn saksopplysinger når man endrer sakstype

janitamarielle committed:
Send kun med behandlingsfrist dersom den er endret

janitamarielle committed:
Logikk for om vi sletter eller oppfrisker trygdeavtale-flyt.
Renamer resetFlyt til oppfriskFlyt, og endret path ++ mot trygdeavtale

janitamarielle commented:
Merk at når du puller [e4fb6f8](https://github.com/navikt/melosys-web/pull/1703/commits/e4fb6f8a45f4960157b5519e58744f633978b555) må du også bytte til å kjøre ny trygdeavtale: https://github.com/navikt/melosys-trygdeavtale/pull/76

janitamarielle committed:
Endrer til ny toggle slik at vi kan slå på toggle når den er klar til prod.

janitamarielle committed:
Dersom anmodningsperiode er sendt til utlandet skal man ikke kunne endre type eller tema på fagsak eller behandling.
Legger ansvaret på å resette anmodningsperiode-state til komponentene som henter de.

frans-a commented:
Ville være greit å fjerne kode for gamle endepunkter for endring av behandlinger siden vi gjorde noe tilsvarende i api

janitamarielle committed:
Fjerner ubrukte endepunkt

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-5285_endre-sakstype-sakstema

 Conflicts:
	src/sider/tomFlyt/behandling.tsx

janitamarielle committed:
Merge feil.
